ouroboros-consensus/docs/report/chapters/future/genesis.tex

```diff
@@ -1499,3 +1500,178 @@ \subsection{Possible optimisations}
+intersection between $O$ and $O'$ is \emph{after} the intersection between $O$
+and $S$, this is trivial:
+
+\begin{center}
+\begin{tikzpicture}[yscale=0.5]
+\draw (0,0) node{$\bullet$} -- ++(1,1) -- ++(3,0) node{$\bullet$} node[right]{$O$};
+\draw (0,0) -- ++(1,-1) -- ++(2,0) node{$\bullet$} node[right]{$S$};
+\draw (1,1) node{$\bullet$} -- ++(1,1) -- ++(3,0) node{$\bullet$} node[right]{$O'$};
+\end{tikzpicture}
+\end{center}
+
+Since we can roll back from $S$ to the intersection point with $O$, we can
+clearly also switch from $S$ to $O'$. If the intersection is further back,
+the situation is a bit more subtle:
+
+\begin{center}
+\begin{tikzpicture}[yscale=0.5]
+\draw (0,0) node{$\bullet$} -- ++(1,1) -- ++(3,0) node{$\bullet$} node[right]{$O$};
+\draw (0,0) -- ++(1,-1) -- ++(2,0) node{$\bullet$} node[right]{$S$};
+\draw (-1,1) node{$\bullet$} -- (0,0);
+\draw (-1,1) -- ++(1,1) -- ++(5,0) node{$\bullet$} node[right]{$O'$};
+\end{tikzpicture}
+\end{center}
+
+By assumption we can switch from $O$ to $O'$. This means that the distance from
+$O$ to the intersection with $O'$ is $k$ blocks or less. Moreover, $O$ must be
+at least as long as $S$ (otherwise $O$ could not be preferred over $S$, and $S$
+would not be suboptimal). This means that the distance from $S$ to the
+intersection with $O'$ must also be at most $k$, and hence we can switch from
+$S$ to $O'$. This means that adoping $S$ does not prevent us from switching to
+any other chain $O'$ that we could have switched to from $O$. \qed
+
+Under exceptional circumstances it is possible that if we \emph{ourselves} forge
+a block on top of a suboptimal chain, the distance from that suboptimal chain to
+the intersecion with the optimal one might now exceed $k$. However, the
+``network respects $\Delta$'' principle saves us from having to treat this as a
+special case.
+
+\subsection{When the node is behind}
+
+To a certain extend, no special provisions need to be made when the node is
+behind. The rule that prohibits adopting more than $k$ blocks from suboptimal
+chains still guarantees that we can continue to switch to the optimal chain
+once we get its blocks; the only difference is that we might hit that limit
+much more quickly.
+
+However, we do have to be careful: when we are behind we cannot rely on the
+property that if we have the right to produce a block in the current slot then
+we must have seen the relevant predecessor to use. We might be able to solve
+this by not producing blocks when we are behind, but this leads to complications
+elsewhere (for example, support for low density chains, \cref{low-density}).
+TODO\todo{TODO}: Unlike what the report says, we in fact decided we \emph{do}
+want to produce blocks when behind, to avoid attackers being able to trick us
+into skipping our slot; we should update that section
+(\cref{sec:block-production-when-behind}) and then refer to it here.
+
+It is therefore probably safer to disable suboptimal chain selection when we
```

Comment:
So would we disable it when being up to date and then falling behind? i.e. could a node adopt at most k blocks of a suboptimal chain, then fall behind (the wallclock goes over the end of the genesis window) and then disable suboptimal chain selection and waiting for the optimal chain to arrive?